packages/js/src/utils/buildable-libs-utils.ts

```diff
@@ -12,17 +12,48 @@ import type * as ts from 'typescript';
 import { unlinkSync } from 'fs';
 import { output } from 'nx/src/utils/output';
 import { isNpmProject } from 'nx/src/project-graph/operators';
-import { ensureTypescript } from './typescript/ensure-typescript';
 import { readTsConfigPaths } from './typescript/ts-config';
 
-let tsModule: typeof import('typescript');
-
-function isBuildable(target: string, node: ProjectGraphProjectNode): boolean {
-  return (
-    node.data.targets &&
-    node.data.targets[target] &&
-    node.data.targets[target].executor !== ''
-  );
+function isBuildable(node: ProjectGraphProjectNode, target: string): string {
+  let buildTargetName: string | undefined = undefined;
+  // TODO(katerina): Remove @nrwl/* for Nx 17
+  const listOfBuildExecutors = [
+    '@nx/js:swc',
+    '@nx/js:tsc',
+    '@nx/rollup:rollup',
+    '@nx/rspack:rspack',
+    '@nx/vite:build',
+    '@nx/esbuild:esbuild',
+    '@nx/webpack:webpack',
+    '@nx/angular:ng-packagr-lite',
+    '@nx/angular:package',
+    '@nx/next:build',
+    '@nrwl/js:swc',
+    '@nrwl/js:tsc',
+    '@nrwl/rollup:rollup',
+    '@nrwl/rspack:rspack',
+    '@nrwl/vite:build',
+    '@nrwl/esbuild:esbuild',
+    '@nrwl/webpack:webpack',
+    '@nrwl/angular:ng-packagr-lite',
+    '@nrwl/angular:package',
+    '@nrwl/next:build',
+  ];
+  for (const targetName of Object.keys(node.data.targets ?? {})) {
+    const target = node.data.targets[targetName];
+    if (listOfBuildExecutors.includes(target.executor)) {
+      buildTargetName = targetName;
+      break;
+    }
+  }
+  if (!buildTargetName) {
+    return node.data.targets &&
+      node.data.targets[target] &&
+      node.data.targets[target].executor !== ''
+      ? target
+      : undefined;
```

Comment:
I would do this first. It's a faster lookup than looping through the targets and a lot of times, folks would have the same target name, so we could determine this pretty quickly and fall back to look if there are targets with known executors.

Reply:
But can we assume that the other target has the same name, and if it does have the same name, it has the same purpose?

Reply:
The order in which it is asserted doesn't change that we still make that assumption, but that's the existing behavior. We wouldn't be breaking anything.

Reply:
Let's keep it the way you have it. You're right that by checking known executors first, we'd guarantee more correctness if there's a match and we fall back to the old checks for backward compat.